### ISSUE TITLE
Callback configuration

### ISSUE THREAD

**Issue by morty** [QuickPay/quickpay-python-client#16]
Having upgraded the version of the QuickPay client in a hurry to handle the TLS 1.0 switch off we have noticed an issue with the callback on payments not being called.

In commit https://github.com/QuickPay/quickpay-python-client/commit/77dc7ba69977d83acef3abc2952129230048150e the client has been changed to add the parameter `callback_url` as a header `Quickpay-Callback-Url`.

This is not working for us currently. The callback URL is not being called and payments are not registering in our system.

I notice that the header is not capitalised in the same way as in the documentation where the p is upper case:

https://learn.quickpay.net/tech-talk/api/callback/

But when I have changed this to match the documentation the callback is still not called.

Because the parameter `callback_url` is popped from the `kwargs` it is not longer passed in the JSON sent in the API call. This is what has caused the callback to stop working for us.

Please can you advise on the best course of action? Should QuickPay-Callback-Url header be working on the V10 API?

**Comment by morty** [comment]
As I said when I raised this issue this fix does not work for me. Even when the header is correctly capitalised the callback is not called.

Please can you confirm that the `QuickPay-Callback-Url` header should be working with the V10 API?

**Comment by Thorbenl** [comment]
Have the same problem right now :/

**Comment by ta** [comment]
@morty @Thorbenl It would very useful if you could provide some example code that reproduces the problem.

I can confirm that `QuickPay-Callback-Url`-header is working as expected for other clients.

@dinesh Could you look into this?

**Comment by dinesh** [comment]
@morty @Thorbenl @ta I did check it again and it seems to be working for me. Are you guys not using the latest version `(1.0.1)`?

Here is an example and the result attached -

```python
from quickpay_api_client import QPClient
client = QPClient(":test_merchant")

response = client.post("/payments", currency="DKK", order_id="order-3")

print client.post("/payments/%s/authorize" % response['id'], 
            amount=300, card={"number": "1000000000000008", "expiration":"3012", "cvd":"123"},
            acquirer='clearhaus',
            callback_url="https://webhook.site/86f4fa63-2d86-41b8-befa-9af28e837bd1")
```

![image](https://user-images.githubusercontent.com/61168/52049166-1bd63280-2573-11e9-8aa7-86715854dfca.png)

**Comment by nickknissen** [comment]
I seem to have found the issue described by @morty. The problem occurs when you try to create a payment link. 
The API expects a `callback_url` in the body and not as a header. 
Because the callback_url is always converted to a header as seen in:
https://github.com/QuickPay/quickpay-python-client/blob/77dc7ba69977d83acef3abc2952129230048150e/quickpay_api_client/api.py#L53-L55

The callback_url won't work for the endpoints which needs the callback_url to be in the body of the request

EDIT:
The following request wont work if callback_url is defined
 - `/account`
 - `/payments/{id}/link`
 - `/payouts/{id}/link`
 - `/subscriptions/{id}/link`

**Comment by ta** [comment]
@dinesh The seems to be the same problem as with the ruby client (1.x.x) where some arguments are magically interpreted as a header.

We should do the same upgrade as ruby client (2.x.x) so that arguments are separated `.method(path, body, headers, ...)` - See https://github.com/QuickPay/quickpay-ruby-client#sending-request

**Comment by nickknissen** [comment]
Is there any ETA on the fixes for the libraries?

**Comment by dinesh** [comment]
@nickknissen added a more generic fix for in #20, hope that it will work for you. We will soon release it with a new version.

**Comment by nickknissen** [comment]
@dinesh When is soon? or should we rely on community forks

**Comment by dinesh** [comment]
@nickknissen released a new version `2.0.0`, please note that the method signature is changed.
